cmd/serverless-init/initcontainer/instrument.go

```diff
@@ -0,0 +1,86 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package initcontainer
+
+import (
+	"os"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/spf13/afero"
+)
+
+// Tracer holds a name, a path to the trace directory, and an
+// initialization function that automatically instruments the
+// tracer
+type Tracer struct {
+	FsPath string
+	InitFn func()
+}
+
+func instrumentNode() {
+	currNodePath := os.Getenv("NODE_PATH")
+	os.Setenv("NODE_PATH", addToString(currNodePath, ":", "/dd_tracer/node/"))
+
+	currNodeOptions := os.Getenv("NODE_OPTIONS")
+	os.Setenv("NODE_OPTIONS", addToString(currNodeOptions, " ", "--require dd-trace/init"))
+}
+
+func instrumentJava() {
+	currJavaToolOptions := os.Getenv("JAVA_TOOL_OPTIONS")
+	os.Setenv("JAVA_TOOL_OPTIONS", addToString(currJavaToolOptions, " ", "-javaagent:/dd_tracer/java/dd-java-agent.jar"))
+}
+
+func instrumentDotnet() {
+	os.Setenv("CORECLR_ENABLE_PROFILING", "1")
+	os.Setenv("CORECLR_PROFILER", "{846F5F1C-F9AE-4B07-969E-05C26BC060D8}")
+	os.Setenv("CORECLR_PROFILER_PATH", "/dd_tracer/dotnet/Datadog.Trace.ClrProfiler.Native.so")
+	os.Setenv("DD_DOTNET_TRACER_HOME", "/dd_tracer/dotnet/")
+}
+
+func instrumentPython() {
+	os.Setenv("PYTHONPATH", addToString(os.Getenv("PYTHONPATH"), ":", "/dd_tracer/python/"))
+}
+
+// AutoInstrumentTracer searches the filesystem for a trace library, and
+// automatically sets the correct environment variables.
+func AutoInstrumentTracer(fs afero.Fs) {
+	tracers := []Tracer{
```

Comment:
nit: Allocations like this are the slowest thing in Go. I'm trying to think of ways to avoid that. One idea is to define the `tracers` on a global level, but instead of a slice, define it as an array.

```go
const tracers =[4]Tracer{
 	{"/dd_tracer/node/", instrumentNode},
	{"/dd_tracer/java/", instrumentJava},
	{"/dd_tracer/dotnet/", instrumentDotnet},
	{"/dd_tracer/python/", instrumentPython},
}
```

I _believe_ this will allocate the memory during compile time instead of during runtime.

Reply:
I think it's _okay_ for now to leave it as is. I can't imagine allocating a slice of size 4 on the heap is ever going to be a bottleneck, and tbh going from 4 -> 5 when we add a new tracer is probably more bothersome.

Reply:
Oh yeah totally. I just can't help myself but to think of all the ways to optimize a piece of code. I think what you have here is the most intuitive.